aviary/cgcnn/data.py

```diff
@@ -288,7 +293,9 @@ def collate_batch(dataset_list):
     batch_cif_ids = []
     base_idx = 0
 
-    for i, (inputs, target, comp, cif_id) in enumerate(dataset_list):
+    # TODO: unpacking (inputs, target, comp, cif_id) doesn't appear to match the doc string
+    # for dataset_list, what about nbr_idx and comp?
+    for idx, (inputs, target, comp, cif_id) in enumerate(dataset_list):
         atom_fea, nbr_dist, self_idx, nbr_idx = inputs
```

Comment:
@CompRhys I tried to type hint the `collate_batch()` functions. Admittedly, it turned into a very long, not very readable annotation. And I'm not even sure, the types are correct. I noticed there appear to be some inconsistencies between the doc strings and the actual input/return types. Left a TODO here that applies to all 3 functions.

Reply:
I attempted to do the typing here but also lost track a little. I think it's a bit clearer but still not clear